Assets/MRTK/SDK/Experimental/Elastic/Scripts/VolumeElasticSystem.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.PlayModeTests")]
+namespace Microsoft.MixedReality.Toolkit.Experimental.Physics
+{
+    public class VolumeElasticSystem : ElasticSystem<Vector3>
+    {
+        // Internal system state.
+        private Vector3 currentValue;
+        private Vector3 currentVelocity;
+
+        // Configuration of extent and spring properties.
+        private VolumeElasticExtent extent;
+        private ElasticProperties elasticProperties;
+
+        public VolumeElasticSystem(Vector3 initialValue, Vector3 initialVelocity,
+                                   VolumeElasticExtent extentInfo,
+                                   ElasticProperties elasticProperties)
+        {
+            currentValue = initialValue;
+            currentVelocity = initialVelocity;
+            this.extent = extentInfo;
+            this.elasticProperties = elasticProperties;
+        }
+
+        /// <inheritdoc/>
+        public override Vector3 ComputeIteration(Vector3 forcingValue, float deltaTime)
+        {
+            // F = -kx - (drag * v)
+            var force = (forcingValue - currentValue) * elasticProperties.HandK - elasticProperties.Drag * currentVelocity;
+
+            // Iterate over each snapping point, and apply forces as necessary.
+            foreach (Vector3 snapPoint in extent.SnapPoints)
+            {
+                Vector3 closestPoint = extent.repeatSnapPoints ? FindNearest(currentValue, snapPoint) : snapPoint;
+                // Calculate distance from snapping point.
+                var distFromSnappingPoint = closestPoint - currentValue;
+                force += ComputeSnapForce(distFromSnappingPoint, elasticProperties.SnapK, extent.SnapRadius);
+            }
+
+            // Check if our current value is within the specified bounds.
+            // If outside, we will apply the end-force (if the bounds are enabled)
+            if (extent.StretchBounds.Contains(currentValue) == false && extent.UseBounds)
```

Comment:
```suggestion
            if (!extent.StretchBounds.Contains(currentValue) && extent.UseBounds)
```

Reply:
Is this not a code style issue? I always thought it was preferred to do `var == false` instead of `!var` because it makes it much more explicit and readable. (This was the policy on my old team, but if we do it differently here, all good)

Reply:
OK. Had a discussion with Will about this, changing now